pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -192,10 +192,17 @@ func buildLogsTimeSeriesFilterQuery(fs *v3.FilterSet, groupBy []v3.AttributeKey,
 					conditions = append(conditions, fmt.Sprintf(logsOp, columnName, fmtVal))
 				case v3.FilterOperatorContains, v3.FilterOperatorNotContains:
 					columnName := getClickhouseColumnName(item.Key)
-					conditions = append(conditions, fmt.Sprintf("%s %s '%%%s%%'", columnName, logsOp, item.Value))
+					conditions = append(conditions, fmt.Sprintf("lower(%s) %s lower('%%%s%%')", columnName, logsOp, item.Value))
 				default:
 					columnName := getClickhouseColumnName(item.Key)
 					fmtVal := utils.ClickHouseFormattedValue(value)
+
+					// for use lower for like and ilike
+					if op == v3.FilterOperatorLike || op == v3.FilterOperatorNotLike {
+						columnName = fmt.Sprintf("lower(%s)", columnName)
```

Comment:
Can you share some examples with and without `lower`? The way I imagined this would work is that we need to make the index lower and convert the search value to lower but do not use `lower` function. Just to double check we will not run into this kind of problem https://stackoverflow.com/questions/62629579/mysql-why-using-where-on-date-column-does-not-use-index

Reply:
This is tested , using `lower(body)` in the where clause uses the corresponding `lower(body) index`
https://github.com/SigNoz/signoz/issues/4259#issuecomment-1970424329

Reply:
Can I play with this somewhere?

Reply:
please check slack